packages/type-utils/src/typeFlagUtils.ts

```diff
@@ -1,31 +1,63 @@
 import { unionTypeParts } from 'tsutils';
 import * as ts from 'typescript';
 
+const ANY_OR_UNKNOWN = ts.TypeFlags.Any | ts.TypeFlags.Unknown;
+
 /**
- * Gets all of the type flags in a type, iterating through unions automatically
+ * Gets all of the type flags in a type, iterating through unions automatically.
  */
-export function getTypeFlags(type: ts.Type): ts.TypeFlags {
-  let flags: ts.TypeFlags = 0;
+export function getTypeFlags(type: ts.Type): number | ts.TypeFlags {
+  let flags = 0;
   for (const t of unionTypeParts(type)) {
     flags |= t.flags;
   }
   return flags;
 }
 
+function isFlagSet(flags: number, flag: number): boolean {
+  return (flags & flag) !== 0;
+}
```

Comment:
This is the same as what `tsutils` does on the inside: https://github.com/ajafff/tsutils/blob/03b4df15d14702f9c7a128ac3fae77171778d613/util/util.ts#L74-L76

```ts
function isFlagSet(obj: {flags: number}, flag: number) {
    return (obj.flags & flag) !== 0;
}
```

There's no need to prefer our own, I think?

Reply:
Aww, `tsutils` doesn't export `isFlagSet`...